### DISCUSSION TITLE
No "IsPresent" method for "WebElements" or "WebElement" object

### DISCUSSION THREAD

**Discussion by GHRyunosuke** [GCuser99/SeleniumVBA#80] [Q&A] [answered]
For example, I narrowed the scope to "WebElements" or "WebElement" objects with some conditions. Then I would like to use "IsPresent" method to check further if some specific element is within the above "WebElements" or "WebElement" object. 

However, I found there is no "IsPresent" method for "WebElements" or "WebElement" objects.

Will the "IsPresent" method be added for "WebElements" or "WebElement" in the future? Or is there an alternative and elegant way to do this (just using "FindElement(s)" would cause a run-time error if does not exist... )?

**Comment by GCuser99** [comment, marked as answer]
@GHRyunosuke, I think adding IsPresent method to WebElement object is a great idea - will do this on the next update (v4.1) coming this weekend!

On adding it to the WebElements object - you will be able to do the following:
```
For Each elem In elems
     If elem.IsPresent([your selector here]) Then 
          'take action
     Else
          'take action
     End If
Next elem
```

**Reply by GHRyunosuke** [reply]
Thank you @GCuser99, that would be great!

**Comment by 6DiegoDiego9** [comment]
Hi @GHRyunosuke 

The current way that I use for testing the equivalent of a
`elem.isPresent(..., ...)`

is
`elem.FindElements(..., ...).Count = 0`
(remember to start with a "." before "/" or "//" if using an XPath)

Anyway, I agree that a new IsPresent method for both WebElement and WebElements objects would be handier, and potentially more efficient (exiting at the first match) :)

**Reply by GHRyunosuke** [reply]
Thank you for the advice, looking at the current "[Object Model Overview](https://github.com/GCuser99/SeleniumVBA/wiki/Object-Model-Overview)" for "WebElement" and "WebElements", I noticed that WebElement is the one having all the "FindElementXXX" methods just as those in "WebDriver". Therefore, adding IsPresent to "WebElement" object (and to WebElement object only) seems to be the best while following the current "design".

**Reply by GCuser99** [reply]
> (remember to start with a "." before "/" or "//" if using an XPath)

I added that advice to Wiki [here](https://github.com/GCuser99/SeleniumVBA/wiki#element-selection-methods)...

**Comment by GCuser99** [comment]
@GHRyunosuke - added to latest release. Also added a test_IsPresent module for usage cases... Thanks again!

**Reply by GHRyunosuke** [reply]
@GCuser99  - Awsome!

One more thing, maybe adding "User can optionally set an implicit wait, and get a reference to the found element." to the wiki of "WebElement	IsPresent" too. It has more than just a boolean :)

![image](https://github.com/GCuser99/SeleniumVBA/assets/25561705/b14d8124-d7bb-4387-aa3e-5efacca08022)

![image](https://github.com/GCuser99/SeleniumVBA/assets/25561705/95adb3c4-8416-47a2-bb39-2794e61786a7)

**Comment by GCuser99** [comment]
Will do - thanks @GHRyunosuke!
